app/assets/javascripts/initializers/initScrolling.js

```diff
@@ -1,26 +1,45 @@
 /* global insertAfter, insertArticles, buildArticleHTML, nextPage:writable, fetching:writable, done:writable, InstantClick */
 
-var client;
+// The purpose of initScrolling is to start paginating the page once you reach the bottom of the page.
+// It does this for majority of the views on the dashboard page as well as the home feed.
 
+var client;
+/**
+ *
+ * @param {String} el is an HTML element that contains the container that we want to append items to. It contains id: indexContainer.
+ * @param {String} endpoint is the relative URL of the endpoint to call to get the data
+ * @param {Function} insertCallback is a function from the return of the callback function (insertNext). The function is passed the data as an argument to build the HTML.
+ * @returns nil but calls the insertCallback (i.e. the insertEntries function) with the data as an argument to update the HTML. It also handles the loading spinner.
+ */
 function fetchNext(el, endpoint, insertCallback) {
   var indexParams = JSON.parse(el.dataset.params);
-  var urlParams = Object.keys(indexParams)
+  // I change the name of the param from "action" to "controller_action" prior to the fetch because Rails ignores the
+  // "action" param in the corresponding endpoint controller because it is a reserved word in Rails. Based on the above
+  // I thought it would be safe to do the param name update for now until we can refactor this file and change the coupled architecture.
+  // "action" renamed to "controller_action" is the name of the action from the originating server side request e.g. hidden_tags in the dashboard_controller.
+  const updatedIndexParams = {};
+  if (indexParams['action']) {
+    delete Object.assign(updatedIndexParams, indexParams, {
+      ['controller_action']: indexParams['action'],
+    })['action'];
+  }
```

Comment:
I'm not _completely_ happy with this change but it made the most sense within the context to be able to pass a useful parameter through to rails. Due to the manner in which initScrolling is written this parameter is passed from the `dashboard` to `followings` to tie the server side rendered GET with the client side pagination fetch.  As per my code comment "action" was already being passed through to rails but never processed. The bigger refactor here would be to rename the variable from the beginning (or architect it a bit differently) so that we're not changing its name just prior to submitting. However, without tests in place and the fact that this file touches so many resources, I felt like this was the less risky change.